### ISSUE TITLE
Blob decoupling

### ISSUE THREAD

**Issue by realbigsean** [sigp/lighthouse#3989]
## Description

We have a few streams of work going on in parallel here so I'm hoping to just concretely split up the tasks to make sure features aren't duplicated and so people know who to reach out to 

Spec PR: https://github.com/ethereum/consensus-specs/pull/3244

- [x] #3991
    - PR: https://github.com/sigp/lighthouse/pull/3950
    - creating a new "observed cache" to filter duplicates 
        - from Pawan: 
        -  > The observed cache has a few more duties than the attestation and sync committee ones
the cache contains incomplete components of a fully available block (i.e AvailabilityPendingBlock + 4 blobs). Once we  get all the components, we essentially go from the AvailabilityPendingBlock to an AvailableBlock which we can process in fork choice
    - create a blob reprocessing flow
    - make the block reprocessing flow dependent on blobs
    - new gossip validation conditions for singular blobs
- [x] #4066
- [x] #4067
- [x] #3994 
    - PR: https://github.com/sigp/lighthouse/pull/3979
    - involves creating a cache in the beacon node for full blobs, and only sending blinded blobs to the VC
- [x] #4099 
- [x] #3992
- [x] #3994
   - Add a new blob signing domain (this should be explicit in the new spec)
   - This will eventually involve updates related to the web3signer but I don't think this will be spec'd for a bit
- [x] #3995
    - branch: https://github.com/realbigsean/lighthouse/tree/decouple-blobs
- [x] Update crypto interface
    -  Issue: https://github.com/sigp/lighthouse/issues/4076
    - PR: https://github.com/sigp/lighthouse/pull/4077
- [x] Add KZG commitment EF tests - https://github.com/sigp/lighthouse/pull/4274

**Comment by jimmygchen** [comment]
Devnet v5 spec: https://hackmd.io/@inphi/HJZo4vQGn

Devnet v5:
- [ ] https://github.com/sigp/lighthouse/issues/3989 (this issue)
- [x] https://github.com/sigp/lighthouse/pull/4186
- [x] https://github.com/ethereum/execution-apis/pull/392 (
  - Engine API change, may not be required if the endpoint gets merged to `getPayloadV3` (below)
- [x] https://github.com/sigp/lighthouse/issues/4210
  - Engine API change, spec PR not merged, tentative for devnet v5
- [x] https://github.com/sigp/lighthouse/issues/3993 (optional)
  - [x] https://github.com/sigp/lighthouse/pull/4075
  - [x] https://github.com/sigp/lighthouse/pull/4083

Nice to have:
- [x] https://github.com/ethereum/beacon-APIs/pull/286
  - [x] https://github.com/sigp/lighthouse/issues/4106
  - [ ] Add blob indices query parameter

Other:
- [x] https://github.com/sigp/lighthouse/issues/4103

**Comment by realbigsean** [comment]
I believe @ethDreamer has started on the execution API related changes [#4120](https://github.com/sigp/lighthouse/issues/4210) and [#392](https://github.com/ethereum/execution-apis/pull/392)

**Comment by realbigsean** [comment]
Going to close this as we've now decoupled in https://github.com/sigp/lighthouse/pull/4054 and I'd like to consolidate clutter wrt tracking
